src/right-field/results-table/ResultsTable.jsx

```diff
@@ -4,180 +4,170 @@ import config from "../../config.json"
+          className={{ tbody: "grid-body" }}
+          data={results}
+          sort={true}
+          autowidth={false}
+          fixedHeader={true}
+          ref={containerRef}
+          columns={variables.map(column => { return generateColumn(column, variables.length) })} />
+      }
+    </div>
+  )
 }
 
+const adderFunctionMapper = {
+  "bindings": (setter) => { return (item, variable) => bindingStreamAdder(item, variable, setter) }
+}
 
 /**
- * A function that given a BindingStream processes every result as a stream based on the functions provided. 
- * @param {Promise<BindingStream>} execution   
- * @param {Function} adder a function which handles what happens with every variable result  
- * @param {Function} variableSetter a function which handles what happens with every variable name 
+ * Processes a result entry as it should look in the result table, and adds it to the table entries
+ * @param {Object} item the entry which should be processed 
+ * @param {Array<String>} variables A list of all the variables in the table 
+ * @param {Function} setter function to set the table entries.
  */
-async function handlequeryExecution(execution, adder, variableSetter){
-  try{
-    let bindingStream = await execution 
-    let variablesMain = []
-    bindingStream.on('data', (binding) => {
-      let variables = []
-      let keys = binding.keys()
-      let key = keys.next()
-      while(!key.done){
-          variables.push(key.value.value)
-          key = keys.next()
-      }   
-      variablesMain = extendList(variablesMain, variables)
-      variableSetter(variablesMain)
-      adder(binding, variablesMain)
-    })
-
-    bindingStream.on('error', handlequeryResultFail)
+function bindingStreamAdder(item, variables, setter) {
+  let newValues = []
+  for (let variable of variables) {
+    let value = item[variable] ? item[variable] : ""
+    let type = variable.split('_')[1]
+    let componentCaller = typeRepresentationMapper[type]
+    componentCaller = componentCaller ? componentCaller : (text) => text.value
+    newValues.push(componentCaller(value))
   }
-  catch(error){
-    handleBindingStreamFail(error)
-  }
-   
+
+  setter(old => { return [...old, newValues] })
 }
 
-function extendList(list, newList){
-  for(let value of newList){
-    if(!list.includes(value)){
-      list.push(value)
+/**
+ * 
+ * @param {Function} adder function which processes the result, takes the result as argument
+ * @param {Function} variableSetter setter function for the variables, takes a list of variable names
+ * @param {Function} setIsQuerying boolean setter function for whether the worker is still executing a query or not
+ */
+function configureQueryWorker(adder, variableSetter, setIsQuerying) {
+  queryWorker = new QueryWorker()
+  let variablesMain = []
+  queryWorker.onmessage = ({ data }) => {
+    switch (data.type) {
+      case 'result':
```

Comment:
Either use single quotes or double quotes, but not both.

Reply:
I have adjusted this in [commit a9ae377](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/a9ae3773cabbc1aed0d43c54d72e5493048f4625)